### ISSUE TITLE
User Permission for Data Collection

### ISSUE THREAD

**Issue by yigitozkavci** [bounswe/bounswe2017group10#405]
We are currently collecting visit data from the users, and saving it in order to improve our recommendation algorithm. We also collect and make use of favorite and commenting data.

Also, we will start to record visit duration on cultural heritage for user and also use it to improve recommendation & search algorithm.

I think we should ask for permission of the user before sending these data to our server. This can be done in both Android and web applications, purely on client-side.

What do you think? @eozd @SaitTalhaNisanci @emir350z

**Comment by YmirKhang** [comment]
Yes, I think we should do it if it will not cause a big load of rework.

**Comment by eozd** [comment]
I don't think client can do anything about data collection from favorites and comments; it is done purely on backend without any additional client requests except the actual favorite/comment create requests.

Regarding visit time, I believe it is a good idea.

**Comment by yigitozkavci** [comment]
You are right @eozd, but I meant the part that we use that data to improve recommendations for "all" users. Collecting the fav and comment data is not the issue, but using it to improve recommendations might be

**Comment by eozd** [comment]
We might add options on which parts of the favorites/comments/... to use in recommendations. Maybe backend can keep user preferences regarding recommendations?

**Comment by yigitozkavci** [comment]
I think that would be an overkill, we can just ask user to give permission or not, and backend can keep track of this with a single boolean associated with user
